fetch_worker.go

```diff
@@ -43,15 +46,36 @@ type fetchWorkerConfig struct {
 	sortColName  string
 }
 
-func newFetchWorker(db *sqlx.DB, data chan fetchData, config fetchWorkerConfig) *fetchWorker {
-	return &fetchWorker{
+func newFetchWorker(db *sqlx.DB, data chan fetchData, config fetchWorkerConfig) fetchWorker {
+	if config.sortColName == "" {
+		return newFetchWorkerByLimit(db, data, config)
```

Comment:
It would be good to log that this or that fetch worker will be used. I feel that it's slightly safer to also do it here, in addition to the logs in source.go, since we can't be 100% sure what happens in-between.:)

Reply:
I'll do that rather in the fetchWorker.run() method, I don't want newFetchWorker with side effects, but I get your point!